packages/@react-aria/tabs/src/TabsKeyboardDelegate.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {Collection, KeyboardDelegate, Node} from '@react-types/shared';
+import {Key} from 'react';
+
+interface TabsKeyboardDelegateOptions<T> {
+  collection: Collection<Node<T>>,
+  disabledKeys: Set<Key>,
+  orientation: 'horizontal' | 'vertical'
+}
+
+
+export class TabsKeyboardDelegate<T> implements KeyboardDelegate {
```

Comment:
looks like this delegate won't do wrapping, should tabs keyboard navigation do wrapping? @devongovett do you know the answer to this?
If so, you can probably borrow the logic from ActionButtons, https://github.com/adobe/react-spectrum/blob/main/packages/@react-aria/actiongroup/src/ActionGroupKeyboardDelegate.ts

Reply:
+1 to the above. Looking at https://www.w3.org/TR/wai-aria-practices-1.2/#keyboard-interaction-20, focus wrapping should be supported IMO.

Reply:
we could also consider merging this into `ListKeyboardDelegate` since it's pretty similar. We'd need to add the `orientation` option, as well as options to disable typeahead and page up/page down I guess.

Reply:
That should work too, I guess we would need to add the existing `shouldFocusWrap` logic in `useSelectableCollection` to `ArrowLeft` and `ArrowRight` right?

Reply:
Thank you!

Reply:
@so99ynoodles If I am interpreting @devongovett comment correctly (feel free to correct me if I am misunderstanding @devongovett ) I think the focus wrapping should solely be handled by `useSelectableCollection` via the `shouldFocusWrap` option. As of now, this keyboard delegate actually handles focus wrapping independently of `useSelectableCollection` due to how `getNextKey` and `getPreviousKey` work (namely https://github.com/adobe/react-spectrum/blob/e56351acc1dbf622d1aa4765da47350381df24cf/packages/%40react-aria/tabs/src/TabsKeyboardDelegate.ts#L84-L86 and https://github.com/adobe/react-spectrum/blob/e56351acc1dbf622d1aa4765da47350381df24cf/packages/%40react-aria/tabs/src/TabsKeyboardDelegate.ts#L94-L96).

Thinking out loud, I think the next change would be to do the following:
1. remove `getNextKey` and `getPreviousKey` from TabsKeyboardDelegate
2. replace `this.getNextKey(key)` with `this.collection.getKeyAfter(key)` and `this.getPreviousKey(key)` with `this.collection.getKeyBefore(key)
3. Add the same `shouldFocusWrap` logic that exists in `useSelectableCollection` for `ArrowUp` and `ArrowDown` to `ArrowLeft` and `ArrowRight`
4. Ugh, some of the directional stuff gets hairy with `getFirstKey` and `getLastKey` here, perhaps useSelectableCollection needs `flipDirection` communicated to it? I'll need to think about this more

Reply:
Heh, well... appears all that was copied from ActionGroupKeyboardDelegate so we'd need to change that one too. Perhaps this could be done as a separate task from this PR? Combine all the keyboard delegates into ListKeyboardDelegate.